examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2120 @@
+# @markdown rearrangement task
+
+
+def get_rotation(sim, object_id):
+    quat = sim.get_rotation(object_id)
+    return np.array(quat.vector).tolist() + [quat.scalar]
+
+
+def init_episode_dict(episode_id, scene_id, agent_pos, agent_rot):
+    episode_dict = {
+        "episode_id": episode_id,
+        "scene_id": "data/scene_datasets/coda/coda.glb",
+        "start_position": agent_pos,
+        "start_rotation": agent_rot,
+        "info": {},
+    }
+    return episode_dict
+
+
+def add_object_details(sim, episode_dict, id, object_template, object_id):
+    object_template = {
+        "object_id": id,
+        "object_template": object_template,
+        "position": np.array(sim.get_translation(object_id)).tolist(),
+        "rotation": get_rotation(sim, object_id),
+    }
+    episode_dict["objects"] = object_template
+    return episode_dict
+
+
+def add_goal_details(sim, episode_dict, object_id):
+    goal_template = {
+        "position": np.array(sim.get_translation(object_id)).tolist(),
+        "rotation": get_rotation(sim, object_id),
+    }
+    episode_dict["goals"] = goal_template
+    return episode_dict
+
+
+# set the number of objects to 1 always for now.
+def build_episode(sim, episode_num, object_id, goal_id):
+    episodes = {"episodes": []}
+    for episode in range(episode_num):
+        agent_state = sim.get_agent(0).get_state()
+        agent_pos = np.array(agent_state.position).tolist()
+        agent_quat = agent_state.rotation
+        agent_rot = np.array(agent_quat.vec).tolist() + [agent_quat.real]
+        episode_dict = init_episode_dict(
+            episode, settings["scene"], agent_pos, agent_rot
+        )
+
+        object_attr = sim.get_object_initialization_template(object_id)
+        object_path = os.path.relpath(
+            os.path.splitext(object_attr.render_asset_handle)[0]
+        )
+
+        episode_dict = add_object_details(
+            sim, episode_dict, 0, object_path, object_id
+        )
+        episode_dict = add_goal_details(sim, episode_dict, goal_id)
+        episodes["episodes"].append(episode_dict)
+
+    return episodes
+
+
+episodes = build_episode(sim, 1, object_id, goal_id)
+
+dataset_content_path = "data/datasets/rearrangement/coda/v1/train/"
+if not os.path.exists(dataset_content_path):
+    os.makedirs(dataset_content_path)
+
+with gzip.open(os.path.join(dataset_content_path, "train.json.gz"), "wt") as f:
+    json.dump(episodes, f)
+
+print(
+    "Dataset written to {}".format(
+        os.path.join(dataset_content_path, "train.json.gz")
+    )
+)
+
+
+# %%
+# @title Dataset class to read the saved dataset in Habitat-API.
+# @markdown To read the saved episodes in Habitat-API, we will extend
+# @markdown the `Dataset` class and the `Episode` base class. It will help
+# @markdown provide all the relevant details about the episode through
+# @markdown a consistent API to all downstream tasks.
+
+# @markdown - We will first create a `RearrangementEpisode` by extending
```

Comment:
This text seems like it's been messily wrapped around by Black, you can make it look better by disabling black on these blocks with fmt:off and fmt:on